source/developers/sts-for-operator.rst

```diff
@@ -0,0 +1,170 @@
+------------
+
+STS for the MinIO Operator requires the following:
+
+- MinIO Operator v5.0.0 or later.
+- The deployment **must** have :ref:`TLS configured <minio-tls>`.
+- :envvar:`OPERATOR_STS_ENABLED` environment variable set to ``on``.
+
+Procedure
+---------
+
+1. Enable STS functionality for the deployment
+   
+   .. code-block:: shell
+      :class: copyable
+
+      kubectl -n minio-operator set env deployment/minio-operator OPERATOR_STS_ENABLED=on
+   
+   - Replace ``minio-operator`` with the namespace for your deployment.
+   - Replace ``deployment/minio-operator`` with the value for your deployment's MinIO Operator.
+
+     You can find the deployment value by running ``kubectl get deployments -n <namespace>``, where you replace ``<namespace>`` with the namespace for the MinIO Operator.
+     Your MinIO Operator namespace is typically ``minio-operator``, though this value can change during install.
+
+2. Ensure an appropriate :ref:`policy <minio-policy>` exists on the MinIO Tenant for the application to use
+
+3. Create a yaml document to add necessary resources: 
+
+   - The :ref:`Service Account <minio-operator-sts-service-account>` in the MinIO Tenant for the application to use.
+
+     For more on service accounts in Kubernetes, see the :kube-docs:`Kubernetes documentation <reference/access-authn-authz/service-accounts-admin/>`.
+   - Create a :ref:`Policy Binding <minio-operator-sts-policy-binding>` linking the application to the MinIO Tenant's policy.
+
+5. Apply the yaml file to create the resources on the deployment
+   
+   .. code-block:: shell
+      :class: copyable
+
+      kubectl apply -k path/to/yaml/file.yaml
+
+6. Use an SDK that supports the ``AssumeRoleWithWebIdentity`` like behavior to send a call from your application to the deployment
+
+   The STS API expects a JWT for the service account to exist in the Kubernetes environment.
+   This is typically in a well known location on the deployment, such as ``/var/run/secrets/kubernetes.io/serviceaccount/token``.
+
+   Some SDKs that support ``AssumeRoleRoleWithWebIdentity`` include:
+
+   - :ref:`Golang <go-sdk>`
+   - :ref:`Java <java-sdk>`
+   - :ref:`JavaScript <javascript-sdk>`
+   - :ref:`.NET <dotnet-sdk>`
+   - :ref:`Python <python-sdk>`
+
+   For examples of using the SDKs to assume a role, see :minio-git:`GitHub <operator/tree/master/examples/kustomization/sts-example/sample-clients>`.
+
+Example Resources
+-----------------
+
+.. _minio-operator-sts-service-account:
+
+Service Account
+~~~~~~~~~~~~~~~
+
+A Service Account is a :kube-docs:`Kubernetes resource type <reference/access-authn-authz/service-accounts-admin/>` that allows an external application to interact with the Kubernetes deployment.
+When linked to a pod, such as through a deployment's ``.spec.spec.serviceAccountName`` field, Kubernetes stores a :abbr:`JWT (JSON Web Token)` for the service account in a well-known location, such as ``/var/run/secrets/kubernetes.io/serviceaccount/token``.
```

Comment:
Rather than save it as a file in the pod file system, k8s mounts it, like this for example

```yaml
apiVersion: v1
kind: Pod
metadata:
  labels:
    app: nginx
  name: nginx-deployment-7fb96c846b-mvjhs
  namespace: default
spec:
  containers:
  - image: nginx:1.14.2
    name: nginx
...
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-vsvfz
      readOnly: true
...
  volumes:
  - name: kube-api-access-vsvfz
    projected:
      defaultMode: 420
      sources:
      - serviceAccountToken:
          expirationSeconds: 3607
          path: token
...
```

```suggestion
When linked to a pod, such as through a deployment's ``.spec.spec.serviceAccountName`` field, Kubernetes mounts a :abbr:`JWT (JSON Web Token)` for the service account in a well-known location, such as ``/var/run/secrets/kubernetes.io/serviceaccount/token``.
```